### PR TITLE
PR: Address small PySide2 compatibility issues

### PR DESCRIPTION
**Pull request by hmaarrfk**
## Description of Changes

- I followed the work done in https://github.com/spyder-ide/spyder/pull/21685
- PySide6 still doesn't work. tested with 6.7.0
- Just trying to help increase the compatibility of Spyder with our software stack (conda-forge + pyside6)

### Issue(s) Resolved

Fixes #

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct:
Mark Harfouche

### CODE REVIEW

File: spyder/plugins/editor/widgets/splitter.py [hunk 1]
```diff
@@ -54,7 +55,11 @@ def __init__(self, parent, main_widget, menu_actions, first=False,
                         Defaults to main_widget.unregister_editorstack() to
                         unregister the EditorStack with the EditorMainWidget.
         """
-        super().__init__(parent, class_parent=main_widget)
+        if PYQT5 or PYQT6:
+            super().__init__(parent, class_parent=main_widget)
+        else:
+            QSplitter.__init__(self, parent)
+            SpyderWidgetMixin.__init__(self, class_parent=main_widget)
```
Comment (ccordoba12) [inline spyder/plugins/editor/widgets/splitter.py:62]:
```suggestion
            SpyderWidgetMixin.__init__(self, class_parent=main_widget)

```

**Review by ccordoba12 — COMMENTED** [review]
Tiny suggestion for you, the rest looks good to me.

**Comment by ccordoba12** [comment]
> PySide6 still doesn't work. tested with 6.7.0

That version is affected by https://github.com/spyder-ide/qtpy/issues/480. What about older versions?

**Comment by hmaarrfk** [comment]
> That version is affected by https://github.com/spyder-ide/qtpy/issues/480. What about older versions?

That workaround is so embedded in my workflows i forgot about that issue we worked on!

**Review by ccordoba12 — APPROVED** [review]
Thanks @hmaarrfk!

**Comment by hmaarrfk** [comment]
PS. still having trouble with PySide6 even for 6.6

I'll continue to investigate over time:

```python
$ QT_API=pyside6 spyder
Could not find the Qt platform plugin "wayland" in ""
Traceback (most recent call last):
  File "/home/mark/miniforge3/envs/dev/bin/spyder", line 17, in <module>
    sys.exit(main())
  File "/home/mark/git/spyder/spyder/app/start.py", line 261, in main
    mainwindow.main(options, args)
  File "/home/mark/git/spyder/spyder/app/mainwindow.py", line 1462, in main
    mainwindow = create_window(MainWindow, app, splash, options, args)
  File "/home/mark/git/spyder/spyder/app/utils.py", line 357, in create_window
    main.setup()
  File "/home/mark/git/spyder/spyder/app/mainwindow.py", line 784, in setup
    PLUGIN_REGISTRY.register_plugin(self, PluginClass,
  File "/home/mark/git/spyder/spyder/api/plugin_registration/registry.py", line 345, in register_plugin
    instance = self._instantiate_spyder5_plugin(
  File "/home/mark/git/spyder/spyder/api/plugin_registration/registry.py", line 187, in _instantiate_spyder5_plugin
    plugin_instance = PluginClass(main_window, configuration=CONF)
  File "/home/mark/git/spyder/spyder/api/plugins/new_api.py", line 330, in __init__
    self._container = container = self.CONTAINER_CLASS(
  File "/home/mark/git/spyder/spyder/plugins/application/container.py", line 83, in __init__
    super().__init__(name, plugin, parent)
  File "/home/mark/git/spyder/spyder/api/widgets/main_container.py", line 119, in __init__
    QWidget.__init__(self, parent)
  File "/home/mark/git/spyder/spyder/api/widgets/mixins.py", line 684, in __init__
    super().__init__()
  File "/home/mark/git/spyder/spyder/api/config/mixins.py", line 293, in __init__
    'A SpyderConfigurationObserver must define a `CONF_SECTION` '
RuntimeError: '__init__' method of object's base class (ApplicationContainer) not called.
(dev) ✘-1 ~
02:59 $ ^C
(dev) ✘-INT ~
03:00 $ ip^C
(dev) ✘-INT ~
03:00 $ mamba list pyside6
# packages in environment at /home/mark/miniforge3/envs/dev:
#
# Name                    Version                   Build  Channel
pyside6                   6.6.2           py310h7a3fa89_1    conda-forge
```

**Comment by ccordoba12** [comment]
It seems similar to the issues you fixed here, i.e. Pyside not supporting `super` im classes that do multiple inheritance.

**Comment by hmaarrfk** [comment]
Yeah..... That's what I thought too. I just couldn't find the offending issue.....

**Pull request merged**